baiji/s3.py

```diff
@@ -380,6 +380,8 @@ def execute(self):
         except S3ResponseError as e:
             if e.status == 403:
                 raise S3Exception("HTTP Error 403: Permission Denied on {}".format(" or ".join([x.uri for x in [self.src, self.dst] if x.is_s3])))
+            elif e.status == 404:
+                raise KeyNotFound("Error copying %s to %s: Source or desitnation bucket doesn't exist" % (self.src.uri, self.dst.uri))
```

Comment:
typo: `destination`

Which call is it that returns 404 when a bucket is not found? This exception is covering a lot of code and it's difficult to tell that this exception text is correct.